lib/modules/datasource/docker/index.ts

```diff
@@ -897,6 +897,26 @@ export class DockerDatasource extends Datasource {
         );
         return this.getTags(registryHost, 'library/' + dockerRepository);
       }
+      // JFrog Artifactory - Retry handling when resolving Docker Official Images
+      // These follow the format of {{registryHost}}{{jFrogRepository}}/library/{{dockerRepository}}
+      if (
+        (err.statusCode === 404 || err.message === PAGE_NOT_FOUND_ERROR) &&
+        err.headers['x-jfrog-version'] !== null &&
+        dockerRepository.split('/').length === 2
```

Comment:
What's an example here? e.g. do you mean `some.artifactory.server/some-library-image-name`?

Reply:
The `/library/` path is only required for [Docker Official Images](https://hub.docker.com/search?q=&image_filter=official).

Examples of these when going direct to Docker Hub include:

- `node`
- `python`
- `redis`
...

When used with JFrog Artifactory virtual repositories, these become:

- `some.artifactory.server/some.artifactory.repository/node`
- `some.artifactory.server/some.artifactory.repository/python`
- `some.artifactory.server/some.artifactory.repository/redis`
- ...

Hence checking for the presence of 2 `/` in the name.

Reply:
To clarify, `dockerRepository` in this case would be `some.artifactory.repository/node` and not the full string as shown above? And there's always a repository in the path (never the root) and the repository will never have a `/` within it?

Reply:
> To clarify, dockerRepository in this case would be some.artifactory.repository/node and not the full string as shown above?

`dockerRepository` on line 905 contains the full string extracted from a Dockerfile FROM statement, so in this case it'd be `some.artifactory.server/some.artifactory.repository/node`

> And there's always a repository in the path (never the root) and the repository will never have a / within it?

That's correct - there must always be be `some.artifactory.repository` used within the Dockerfile FROM statement.

Here is a screengrab of the jfrog.io `set me up` screen for Docker
<img width="856" alt="Screen Shot 2022-09-25 at 06 02 16 " src="https://user-images.githubusercontent.com/386277/192138163-a6943ff1-28a2-4eea-bdb2-9937944d2e73.png">

`some.artifactory.repository` will never have a `/` in it.

The `image` however might, but for those instances the current Renovate logic works perfectly with the JFrog Docker Tags List API